src/main/java/com/zendesk/maxwell/row/FieldNameStrategy.java

```diff
@@ -0,0 +1,49 @@
+package com.zendesk.maxwell.row;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * 
+ * @author frankchen Dec 24, 2019 9:08:41 PM
+ */
+public class FieldNameStrategy {
+
+	public final static String UNDERSCORE_TO_CAMEL_CASE = "underscore_to_camelcase";
+
+	// cache converted names
+	private Map<String, String> caches = new ConcurrentHashMap<>();
+
+	private String strategyName;
+
+	public FieldNameStrategy(String strategy) {
+		this.strategyName = strategy;
+	}
+	
+	public String apply(String oldName) {
+		if ( UNDERSCORE_TO_CAMEL_CASE.equals(strategyName)) 
+			return caches.computeIfAbsent(oldName, k -> underscore2Camel(k));
+		
+		return oldName;
+	}
+	
+	private String underscore2Camel(String oldName) {
+		StringBuilder newName = new StringBuilder(oldName.length());
+		for (int i = 0, len = oldName.length(); i < len; i++) {
+			char c = oldName.charAt(i);
+			if (c == '_' ) {
+				if ( newName.length() > 0 //ignore the leading underscore 
+				  && i + 1 < len //this underscore is not the last character 
+				  ) {
+					char afterUnderscore = oldName.charAt(++i);
+					
+					newName.append(Character.isLowerCase(afterUnderscore)	? (char) (afterUnderscore - 'a' + 'A')
```

Comment:
how would this work with non-ascii charaters?  Do they simply fail the `isLowerCase` check?  can you add a test around it?

Reply:
there're several ways to handle non-ascii characters,
1. discard all invalid characters for a variable name. 
2. keep them, leave them to user.
3. replace them one by one according some fixed rules

There're no perfect ways to handle it. if we discard invalid characters, there's some cases which will fail. `...` for example, is a valid name for a table or a column in mysql, after discarding, it turns an empty string.

JOOQ replaces `.` with underscore, and `!` with `_21`. if we apply some rules like that, it's too complex.

MyBatis replaces all underscore with empty chars,
````
  org.apache.ibatis.reflection.MetaClass

  public String findProperty(String name, boolean useCamelCaseMapping) {
    if (useCamelCaseMapping) {
      name = name.replace("_", "");
    }
    return findProperty(name);
  }
````

So my suggestion is to keep those characters, leave them to user. In code, user can still use `JsonProperty` annotation to map the field name in json to a rightful variable name.